Assets/Editor/CreatureHydrator.cs

```diff
@@ -0,0 +1,263 @@
+using YamlDotNet.Serialization;
+using YamlDotNet.Serialization.NamingConventions;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+public class CreatureHydrator : Editor
+{
+    [MenuItem("Tools/Hydrate Creatures")]
+    public static void HydrateCreatures()
+    {
+        string rootGeneraPath = "Assets/Creatures/impl";
+        string[] generaDirectories = Directory.GetDirectories(rootGeneraPath);
+
+        foreach (var generaDir in generaDirectories)
+        {
+            string generaName = Path.GetFileName(generaDir);
+            string abstractClassPath = Path.Combine(generaDir, $"{generaName}.cs");
+
+            if (!File.Exists(abstractClassPath))
+            {
+                Debug.LogError($"Abstract class for {generaName} not found at {abstractClassPath}");
+                continue;
+            }
+
+            string[] unitDirectories = Directory.GetDirectories(generaDir);
+
+            foreach (var unitDir in unitDirectories)
+            {
+                string unitName = Path.GetFileName(unitDir);
+                string unitYamlPath = Path.Combine(unitDir, $"{unitName}.yml");
+
+                if (!File.Exists(unitYamlPath))
+                {
+                    Debug.LogError($"YAML file for unit {unitName} not found at {unitYamlPath}");
+                    continue;
+                }
+
+                string yamlContent = File.ReadAllText(unitYamlPath);
+                UnitDefinition unit = DeserializeYaml<UnitDefinition>(yamlContent);
+
+                // Hydrate the creature using the spec and directory
+                HydrateCreature(unit, generaName, unitName, unitDir);
+            }
+        }
+    }
+
+    private static T DeserializeYaml<T>(string yamlContent)
+    {
+        var deserializer = new DeserializerBuilder()
+            .WithNamingConvention(CamelCaseNamingConvention.Instance)
+            .Build();
+
+        return deserializer.Deserialize<T>(yamlContent);
+    }
+
+
+    private static void HydrateCreature(UnitDefinition unit, string generaName, string unitName, string unitDir)
+    {
+        // Create the Prefab
+        GameObject unitPrefab = new GameObject(unitName);
+        SpriteRenderer spriteRenderer = unitPrefab.AddComponent<SpriteRenderer>();  // We'll set the sprite later
+        var animator = unitPrefab.AddComponent<Animator>();
+
+        // Create an Animator Controller within the unit's directory
+        string animatorPath = Path.Combine(unitDir, $"{unitName}Controller.controller");
+        AnimatorController animatorController = AnimatorController.CreateAnimatorControllerAtPath(animatorPath);
+        
+        //Initialize idleSprite so we can set it from the Idle animation sheet
+        Sprite idleSprite = null;
+
+        // Process each animation
+        foreach (var animation in unit.animations)
+        {
+            string animationFolderPath = Path.Combine(unitDir, animation.name);
```

Comment:
I think some of these might be worth dividing into functions, but don't know if it's all that necessary

Reply:
okay we'll leave it for now